internal/relay/relayer.go

```diff
@@ -78,7 +91,21 @@ func (r *Relayer) Run(ctx context.Context, tasks <-chan neutrontypes.RegisteredQ
 				err = r.processMessageKV(ctx, msg)
 			case string(neutrontypes.InterchainQueryTypeTX):
 				msg := &MessageTX{QueryId: query.Id, TransactionsFilter: query.TransactionsFilter}
-				err = r.processMessageTX(ctx, msg)
+				if r.hasProcessingTxQuery(query.Id) {
+					r.logger.Info("query already in processing", zap.Uint64("query_id", query.Id))
+					continue
+				}
+				x := make(chan error)
+				r.addProcessingTxQuery(query.Id)
+				go (func() {
+					defer r.removeProcessingTxQuery(query.Id)
+					err := r.processMessageTX(ctx, msg)
+					x <- err
+				})()
+				select {
+				case err = <-x:
+				}
```

Comment:
What's the point of running `processMessageTX` in another goroutine if the whole thread is blocked by `select` on line 106 waiting for the `processMessageTX` to be completed via `x` chan?